samples/drumthumper/src/main/cpp/DrumPlayerJNI.cpp

```diff
@@ -80,12 +80,14 @@ JNIEXPORT jboolean JNICALL Java_com_plausiblesoftware_drumthumper_DrumPlayer_loa
     WavStreamReader reader(&stream);
     reader.parse();
 
-    jboolean isFormatValid =
-            (reader.getSampleRate() == rate) && (reader.getNumChannels() == channels);
+    jboolean isFormatValid = reader.getNumChannels() == channels;
 
     SampleBuffer* sampleBuffer = new SampleBuffer();
     sampleBuffer->loadSampleData(&reader);
 
+    int sampleRate = sDTPlayer.getDeviceSampleRate(2 /*channelCount*/);
```

Comment:
Why does getDeviceSampleRate() take a channelCount?
Also should it be channels instead of "2"?

Reply:
The (copy-and-paste) AudioStream builder code takes a channel count. Perhaps it isn't necessary?
If it IS necessary, the 2 is the number of channels being played even though there is only one channel in the sample buffer.